### DISCUSSION TITLE
Is there an example of using vike + telefunc + jest?

### DISCUSSION THREAD

**Discussion by jasonhilldm** [vikejs/vike#2024] [Discussion]
I have a test project that uses vike and telefunc and I am trying to set up some automated tests using jest and @testing-library/react. The test is failing with **Cannot find module './+Page.telefunc.js' from 'pages/public/index/+Page.tsx'**.

Are there any examples of how to setup jest when using vike and telefunc? Are there any special considerations?

**Comment by brillout** [comment]
I ain't aware of a recent Jest example (most use Vitest nowadays).

**Reply by jasonhilldm** [reply]
Ok…will check that out. Should telefunc play nicely with that?

**Reply by brillout** [reply]
> Should telefunc play nicely with that?

Try and report back if you find issues. (See https://brillout.github.io/rules and https://vike.dev/faq#github.)

**Reply by jasonhilldm** [reply]
Just to clarify…are you saying that my last question wasn’t appropriate?

**Reply by brillout** [reply]
I've a lot on my plate, so I've to be careful with my time. It's only that :)

It works best when users try before asking. Otherwise I'll spend too much time helping users. Thank you for understanding.

**Reply by jasonhilldm** [reply]
Yep I understand. That’s why I framed my question as a simple yes/no…if you knew it wasn’t going to play nice then it would save me hours of fumbling around trying to get it working!

**Reply by jasonhilldm** [reply]
I tried using vitest with vike & vike-node but the server keeps restarting: https://github.com/vikejs/vike-node/issues/36
